### PR TITLE
Use StatefulSets for controller-managers and support ShortName for CRD

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

File: cmd/internal/codegen/parse/index.go [hunk 1]
```diff
@@ -71,8 +70,9 @@ func (b *APIs) parseIndex() {
 
 		rt := parseResourceTag(b.getResourceTag(c))
 		r.Resource = rt.Resource
-		r.REST = rt.REST
-		r.Strategy = rt.Strategy
+		r.ShortName = rt.ShortName
+		//r.REST = rt.REST
```
Comment (seans3) [inline cmd/internal/codegen/parse/index.go]: Do you want to check in commented code?

File: cmd/internal/codegen/parse/index.go [hunk 2]
```diff
@@ -96,85 +96,86 @@ func (b *APIs) parseIndex() {
 		b.ByGroupKindVersion[r.Group][r.Kind][r.Version] = r
 		b.ByGroupVersionKind[r.Group][r.Version][r.Kind] = r
 
-		if !HasSubresource(c) {
-			continue
-		}
+		//if !HasSubresource(c) {
```
Comment (seans3) [inline cmd/internal/codegen/parse/index.go]: Ditto.

File: cmd/kubebuilder/create/resource/run.go [hunk 3]
```diff
@@ -82,9 +83,10 @@ func RunCreateResource(cmd *cobra.Command, args []string) {
 			"Regenerate after editing resources files by running `kubebuilder build generated`.\n")
 		build.RunGenerate(cmd, args)
 	}
-	fmt.Printf("Next: Install API and run controller with " +
-		"`GOBIN=$(pwd)/bin go install <your-project-go-package>/cmd/controller-manager`.\n" +
-		"bin/controller-manager --kubeconfig ~/.kube/config\n")
+	fmt.Printf("Next: Install the API, run the controller and create an instance with:\n" +
+		"$ GOBIN=$(pwd)/bin go install <your-project-go-package>/cmd/controller-manager\n" +
+		"$ bin/controller-manager --kubeconfig ~/.kube/config\n" +
+		"$ kubectl apply -f hack/sample/" + strings.ToLower(createutil.KindName) + ".yaml\n")
```
Comment (seans3) [inline cmd/kubebuilder/create/resource/run.go]: The apply example is helpful.

File: cmd/kubebuilder/create/config/gen.go [hunk 4]
```diff
@@ -185,15 +176,104 @@ func getDeployment(p *parse.APIs) string {
...
+			TerminationGracePeriodSeconds: &terminationPeriod,
+			Containers: []corev1.Container{
+				{
+					Name:    "controller-manager",
+					Image:   controllerImage,
+					Command: []string{"/root/controller-manager"},
+					Args:    []string{"--install-crds=false"},
+					Resources: corev1.ResourceRequirements{
+						Requests: map[corev1.ResourceName]resource.Quantity{
+							"cpu":    resource.MustParse("100m"),
```
Comment (seans3) [inline cmd/kubebuilder/create/config/gen.go]: Should these hard-coded values (e.g. "100m") be more configurable?

Comment (pwittrock) [reply cmd/kubebuilder/create/config/gen.go]: This is fine for now.  Can ask Ken what reasonable defaults should be.  Can use kinflate to override them.

**Pull request merged**